app/models/annotation/nml/NmlParser.scala

```diff
@@ -210,17 +211,46 @@ object NmlParser extends LazyLogging with ProtoGeometryImplicits with ColorGener
         case _                           => None
       }
       val anchorPositionAdditionalCoordinates = parseAdditionalCoordinateValues(node)
+      val metadata = parseMetadata(node \ "metadata" \ "metadataEntry")
```

Comment:
Yes, `node \ "metadata" \ "metadataEntry"` returns an empty list if either level is not present. So that should be fine. In fact, the backend nml writer also does not write the metadata tag if it would be empty.

> The frontend nml serializer does not serialize segments.

Correct, the frontend NML writer is called from a context menu action called “export trees”. It does not include the volume layers at all.